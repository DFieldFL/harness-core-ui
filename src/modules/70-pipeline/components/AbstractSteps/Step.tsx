import type { IconName } from '@wings-software/uicore'
import type { FormikErrors } from 'formik'
import type { CompletionItemInterface } from '@common/interfaces/YAMLBuilderProps'
import type { UseStringsReturn } from 'framework/exports'
import type { AbstractStepFactory } from './AbstractStepFactory'
import type { StepType } from '../PipelineSteps/PipelineStepInterface'

export enum StepViewType {
  InputSet = 'InputSet',
  InputVariable = 'InputVariable',
  DeploymentForm = 'DeploymentForm',
  StageVariable = 'StageVariable',
  Edit = 'Edit'
}

export interface InputSetData<T> {
  template?: T
  allValues?: T
  path: string
  readonly?: boolean
}

export interface StepProps<T, U = unknown> {
  initialValues: T
  onUpdate?: (data: T) => void
  stepViewType?: StepViewType
  inputSetData?: InputSetData<T>
  factory?: AbstractStepFactory
  path: string
  customStepProps?: U
}

export abstract class Step<T> {
  protected abstract type: StepType
  protected abstract defaultValues: T
  protected abstract stepIcon: IconName
  protected abstract stepName: string
  protected invocationMap?: Map<
    RegExp,
    (path: string, yaml: string, params: Record<string, unknown>) => Promise<CompletionItemInterface[]>
  >
  abstract validateInputSet(data: T, template?: T, getString?: UseStringsReturn['getString']): FormikErrors<T>

  protected stepPaletteVisible?: boolean // default to true

  getType(): string {
    return this.type
  }

  getDefaultValues(initialValues: T, _stepViewType: StepViewType): T {
    return { ...this.defaultValues, ...initialValues }
  }

  getIconName(): IconName {
    return this.stepIcon
  }

  getStepName(): string {
    return this.stepName
  }

  getInvocationMap():
    | Map<RegExp, (path: string, yaml: string, params: Record<string, unknown>) => Promise<CompletionItemInterface[]>>
    | undefined {
    return this.invocationMap
  }

  getStepPaletteVisibility(): boolean {
    return this.stepPaletteVisible ?? true
  }

  abstract renderStep(props: StepProps<T>): JSX.Element
}
