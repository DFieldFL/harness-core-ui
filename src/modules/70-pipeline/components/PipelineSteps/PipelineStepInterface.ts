export enum StepType {
  HTTP = 'Http',
  SHELLSCRIPT = 'ShellScript',
  APPROVAL = 'Approval',
  K8sRollingRollback = 'K8sRollingRollback',
  StepGroup = 'StepGroup',
  KubernetesInfraSpec = 'KubernetesInfraSpec',
  K8sRollingDeploy = 'K8sRollingDeploy',
  CustomVariable = 'Custom_Variable',
  Redis = 'service',
  Plugin = 'plugin',
  Run = 'run',
  SaveCache = 'Save_Cache',
  RestoreCache = 'Restore_Cache',
  GCR = 'GCR'
}
