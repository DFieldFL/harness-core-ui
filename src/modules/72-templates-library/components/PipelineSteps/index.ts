import factory from '@pipeline/components/PipelineSteps/PipelineStepFactory'

import { TemplateStep } from './TemplateStep/TemplateStep'

factory.registerStep(new TemplateStep())
