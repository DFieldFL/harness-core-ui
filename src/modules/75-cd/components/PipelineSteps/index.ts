import factory from '@pipeline/components/PipelineSteps/PipelineStepFactory'

import { HttpStep } from './HttpStep/HttpStep'
import { K8RolloutDeployStep } from './K8sRolloutDeployStep/K8sRolloutDeployStep'
import { ShellScriptStep } from './ShellScriptStep/ShellScriptStep'
import { KubernetesInfraSpec } from './KubernetesInfraSpec/KubernetesInfraSpec'
import { KubernetesServiceSpec } from './K8sServiceSpec/K8sServiceSpec'
import { K8sBlueGreenDeployStep } from './K8sBgStep/K8sBlueGreenDeployStep'
import { K8sCanaryDeployStep } from './K8sCanaryDeploy/K8sCanaryDeployStep'
import { K8sBGSwapServices } from './K8sBGSwapServices/K8sBGSwapServices'
import { K8sScaleStep } from './K8sScale/K8sScaleStep'
import { K8sRollingRollbackStep } from './K8sRollingRollback/K8sRollingRollback'
import { K8sCanaryDeleteStep } from './K8sCanaryDelete/K8sCanaryDeleteStep'
import { K8sApplyStep } from './K8sApply/K8sApplyStep'
import { K8sDeleteStep } from './K8sDelete/K8sDeleteStep'
import { DeployEnvironmentStep } from './DeployEnvStep/DeployEnvStep'
import { DeployServiceStep } from './DeployServiceStep/DeployServiceStep'
import { HelmDeploy } from './HelmDeploy/HelmDeploy'
import { HelmRollback } from './HelmRollback/HelmRollback'
import { TerraformRollback } from './TerraformRollback/TerraformRollback'
import { TerraformDestroy } from './TerraformDestroy/TerraformDestroy'
import { TerraformPlan } from './TerraformPlan/TerraformPlan'

factory.registerStep(new HttpStep())
factory.registerStep(new K8RolloutDeployStep())
factory.registerStep(new K8sRollingRollbackStep())
factory.registerStep(new K8sBlueGreenDeployStep())
factory.registerStep(new K8sCanaryDeployStep())
factory.registerStep(new K8sBGSwapServices())
factory.registerStep(new K8sScaleStep())
factory.registerStep(new K8sCanaryDeleteStep())
factory.registerStep(new K8sApplyStep())
factory.registerStep(new K8sDeleteStep())
factory.registerStep(new ShellScriptStep())
factory.registerStep(new KubernetesInfraSpec())
factory.registerStep(new DeployEnvironmentStep())
factory.registerStep(new DeployServiceStep())
factory.registerStep(new KubernetesServiceSpec())
factory.registerStep(new HelmDeploy())
factory.registerStep(new HelmRollback())
factory.registerStep(new TerraformRollback())
factory.registerStep(new TerraformDestroy())
factory.registerStep(new TerraformPlan())
