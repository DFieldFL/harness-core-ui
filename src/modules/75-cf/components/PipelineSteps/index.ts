import factory from '@pipeline/components/PipelineSteps/PipelineStepFactory'
import { FlagConfigurationStep } from './FlagConfigurationStep/FlagConfigurationStep'

export function registerFlagConfigurationPipelineStep(): void {
  factory.registerStep(new FlagConfigurationStep())
}
