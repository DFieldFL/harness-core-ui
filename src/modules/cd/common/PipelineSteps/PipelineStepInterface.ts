export enum StepType {
  HTTP = 'Http',
  SHELLSCRIPT = 'ShellScript',
  APPROVAL = 'Approval',
  StepGroup = 'StepGroup',
  K8sRolloutDeploy = 'K8sRolloutDeploy'
}
