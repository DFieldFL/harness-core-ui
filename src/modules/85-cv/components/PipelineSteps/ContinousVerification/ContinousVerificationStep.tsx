import React from 'react'
import type { IconName, SelectOption } from '@wings-software/uicore'
import { omitBy, isNil } from 'lodash-es'
import { StepViewType, StepProps } from '@pipeline/exports'
import type { CompletionItemInterface } from '@common/interfaces/YAMLBuilderProps'
import { StepType } from '@pipeline/components/PipelineSteps/PipelineStepInterface'
import { PipelineStep } from '@pipeline/components/PipelineSteps/PipelineStep'
import type { VariableResponseMapValue } from 'services/pipeline-ng'

import type { ContinousVerificationData, ContinousVerificationFormData, spec } from './continousVerificationTypes'
import { ContinousVerificationWidgetWithRef } from './ContinousVerificationWidget'
import { ContinousVerificationInputSetStep } from './ContinousVerificationInputSetStep'
import { ContinousVerificationVariableStep } from './ContinousVerificationVariableStep'
import {
  baseLineOptions,
  durationOptions,
  trafficSplitPercentageOptions,
  VerificationSensitivityOptions
} from './ConfigureVerificationJob'

const ConnectorRefRegex = /^.+step\.spec\.executionTarget\.connectorRef$/

export interface ContinousVerificationVariableStepProps {
  metadataMap: Record<string, VariableResponseMapValue>
  stageIdentifier: string
  variablesData: ContinousVerificationData
  originalData: ContinousVerificationData
}

export class ContinousVerificationStep extends PipelineStep<ContinousVerificationData> {
  constructor() {
    super()
    this.invocationMap.set(ConnectorRefRegex, this.getSecretsListForYaml.bind(this))
    //TODO will be turned true when variables view story is picked up
    this._hasStepVariables = false
  }

  renderStep(props: StepProps<ContinousVerificationData>): JSX.Element {
    const { initialValues, onUpdate, stepViewType, inputSetData, formikRef, customStepProps } = props

    if (stepViewType === StepViewType.InputSet || stepViewType === StepViewType.DeploymentForm) {
      return (
        <ContinousVerificationInputSetStep
          initialValues={this.getInitialValues(initialValues)}
          onUpdate={data => onUpdate?.(this.processFormData(data))}
          stepViewType={stepViewType}
          readonly={!!inputSetData?.readonly}
          template={inputSetData?.template}
          path={inputSetData?.path || ''}
        />
      )
    } else if (stepViewType === StepViewType.InputVariable) {
      return (
        <ContinousVerificationVariableStep
          {...(customStepProps as ContinousVerificationVariableStepProps)}
          originalData={initialValues}
        />
      )
    }
    return (
      <ContinousVerificationWidgetWithRef
        initialValues={this.getInitialValues(initialValues)}
        onUpdate={data => onUpdate?.(this.processFormData(data))}
        stepViewType={stepViewType}
        ref={formikRef}
      />
    )
  }

  validateInputSet(): object {
    const errors = {} as any
    //TODO implement this once story is picked up

    return errors
  }

  protected type = StepType.Verify
  protected stepName = 'CV Step'
  protected stepIcon: IconName = 'cv-main'
  //TODO - to be asked from PM
  protected isHarnessSpecific = true
  protected invocationMap: Map<
    RegExp,
    (path: string, yaml: string, params: Record<string, unknown>) => Promise<CompletionItemInterface[]>
  > = new Map()

  protected defaultValues: ContinousVerificationData = {
    identifier: '',
    timeout: '10m',
    spec: {
      verificationJobRef: '',
      type: '',
      spec: {
        sensitivity: '',
        duration: '',
        baseline: '',
        trafficsplit: '',
        serviceRef: '',
        envRef: '',
        deploymentTag: ''
      }
    }
  }

  protected async getSecretsListForYaml(): Promise<CompletionItemInterface[]> {
    //TODO implement this once story is picked up
    return new Promise(resolve => {
      resolve([])
    })
  }

  private getInitialValues(initialValues: ContinousVerificationData): ContinousVerificationFormData {
    return {
      ...initialValues,
      spec: {
        ...initialValues.spec,
        verificationJobRef: initialValues?.spec?.verificationJobRef,
        type: initialValues.spec?.type,
        spec: this.getspecFormData(initialValues?.spec?.spec)
      }
    }
  }

  processFormData(data: ContinousVerificationFormData): ContinousVerificationData {
    return {
      ...data,
      spec: {
        ...data.spec,
        verificationJobRef: (data.spec.verificationJobRef as SelectOption)?.value as string,
        type: data.spec?.type,
        spec: this.getspecYamlData(data.spec.spec)
      }
    }
  }

  private getspecYamlData(specInfo: spec | undefined): spec {
    const validspec = omitBy(specInfo, isNil)

    Object.keys(validspec).map((key: string) => {
      validspec[key] = validspec[key].value ? validspec[key].value : validspec[key]
    })

    return validspec
  }

  private getspecFormData(specInfo: spec | undefined): spec {
    const validspec: spec | undefined = { ...specInfo }
    if (specInfo) {
      Object.keys(specInfo).map((key: string) => {
        switch (key) {
          case 'sensitivity':
            this.setFieldData(validspec, 'sensitivity', VerificationSensitivityOptions)
            break
          case 'duration':
            this.setFieldData(validspec, 'duration', durationOptions)
            break
          case 'baseline':
            this.setFieldData(validspec, 'baseline', baseLineOptions)
            break
          case 'trafficsplit':
            this.setFieldData(validspec, 'trafficsplit', trafficSplitPercentageOptions)
            break
          default:
        }
      })
    }
    return validspec
  }

  private setFieldData(validspec: spec | undefined, field: string, options: SelectOption[]): void {
    if (validspec && validspec[field]) {
      validspec[field] = options.find((el: SelectOption) => el.value === (validspec && validspec[field]))
    }
  }
}
