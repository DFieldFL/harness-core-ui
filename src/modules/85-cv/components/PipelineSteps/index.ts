import factory from '@pipeline/components/PipelineSteps/PipelineStepFactory'

import { ContinousVerificationStep } from './ContinousVerification/ContinousVerificationStep'

factory.registerStep(new ContinousVerificationStep())
