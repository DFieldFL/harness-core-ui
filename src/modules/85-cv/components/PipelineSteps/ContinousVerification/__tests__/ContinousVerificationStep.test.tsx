import React from 'react'
import { fireEvent, queryByAttribute, render, waitFor } from '@testing-library/react'
import { RUNTIME_INPUT_VALUE } from '@wings-software/uicore'
import { StepViewType, StepFormikRef } from '@pipeline/components/AbstractSteps/Step'
import { StepType } from '@pipeline/components/PipelineSteps/PipelineStepInterface'
import { factory, TestStepWidget } from '@pipeline/components/PipelineSteps/Steps/__tests__/StepTestUtil'
import { ContinousVerificationStep } from '../ContinousVerificationStep'

const jobsData = {
  data: [
    {
      identifier: 'test11212',
      jobName: 'test-11212',
      serviceIdentifier: '<+input>',
      serviceName: null,
      envIdentifier: '<+input>',
      envName: null,
      projectIdentifier: 'test1',
      orgIdentifier: 'default',
      activitySourceIdentifier: 'cd_nextgen_activity_source',
      dataSources: null,
      monitoringSources: ['AppDynamics'],
      verificationJobUrl:
        '/cv/api/verification-job?accountId=kmpySmUISimoRrJL6NL73w&orgIdentifier=default&projectIdentifier=test1&identifier=test11212',
      duration: '5m',
      sensitivity: 'HIGH',
      trafficSplitPercentage: 5,
      type: 'BLUE_GREEN',
      defaultJob: false
    },
    {
      identifier: 'test1_Built-in_health_verification',
      jobName: 'Built-in health verification',
      serviceIdentifier: '<+input>',
      serviceName: null,
      envIdentifier: '<+input>',
      envName: null,
      projectIdentifier: 'test1',
      orgIdentifier: 'default',
      activitySourceIdentifier: null,
      dataSources: ['APP_DYNAMICS', 'SPLUNK', 'STACKDRIVER', 'KUBERNETES', 'NEW_RELIC'],
      monitoringSources: ['ALL'],
      verificationJobUrl:
        '/cv/api/verification-job?accountId=kmpySmUISimoRrJL6NL73w&orgIdentifier=default&projectIdentifier=test1&identifier=test1_Built-in_health_verification',
      duration: '15m',
      type: 'HEALTH',
      defaultJob: true
    }
  ]
}

jest.mock('services/cv', () => ({
  useCDNGVerificationJobs: jest.fn().mockImplementation(() => ({ loading: false, data: jobsData, error: false }))
}))

jest.mock('@common/components/YAMLBuilder/YamlBuilder', () => ({ children }: { children: JSX.Element }) => (
  <div>{children}</div>
))

describe('Test ContinousVerificationStep Step', () => {
  beforeEach(() => {
    factory.registerStep(new ContinousVerificationStep())
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  test('should render edit view as new step', () => {
    const { container } = render(
      <TestStepWidget initialValues={{}} type={StepType.Verify} stepViewType={StepViewType.Edit} />
    )

    expect(container).toMatchSnapshot()
  })

  test('should render edit view as edit step', () => {
    const initialValues = {
      type: 'ContinousVerification',
      identifier: 'ContinousVerification',
      name: 'CV',
      spec: {
        verificationJobRef: 'Blue Green Job',
        type: 'BLUE_GREEN',
        spec: {
          sensitivity: 'Low',
          duration: '15min',
          baseline: 'Last successful job run',
          trafficsplit: '5%',
          service: 'docerservice',
          env: 'preprod',
          deploymentTag: '1.2'
        }
      }
    }
    const { container } = render(
      <TestStepWidget initialValues={initialValues} type={StepType.Verify} stepViewType={StepViewType.Edit} />
    )

    expect(container).toMatchSnapshot()
  })

  test('renders input sets', () => {
    const onUpdate = jest.fn()
    const initialValues = {
      type: 'ContinousVerification',
      identifier: 'ContinousVerification',
      name: 'CV',
      spec: {
        verificationJobRef: 'Blue Green Job',
        type: 'BLUE_GREEN',
        spec: {
          sensitivity: RUNTIME_INPUT_VALUE,
          duration: RUNTIME_INPUT_VALUE,
          baseline: RUNTIME_INPUT_VALUE,
          trafficsplit: RUNTIME_INPUT_VALUE,
          service: RUNTIME_INPUT_VALUE,
          env: RUNTIME_INPUT_VALUE,
          deploymentTag: RUNTIME_INPUT_VALUE
        }
      }
    }
    const { container } = render(
      <TestStepWidget
        initialValues={{}}
        template={initialValues}
        type={StepType.Verify}
        stepViewType={StepViewType.InputSet}
        onUpdate={onUpdate}
        path=""
      />
    )

    expect(container).toMatchSnapshot()
  })

  test('renders empty input sets', () => {
    const { container } = render(
      <TestStepWidget
        initialValues={{}}
        type={StepType.Verify}
        stepViewType={StepViewType.InputSet}
        template={{}}
        path=""
      />
    )

    expect(container).toMatchSnapshot()
  })

  test('shows different interface when the api call to fetch CV jobs is completed', async () => {
    const onUpdate = jest.fn()
    const ref = React.createRef<StepFormikRef<unknown>>()
    const { container, getByText } = render(
      <TestStepWidget
        initialValues={{}}
        type={StepType.Verify}
        stepViewType={StepViewType.Edit}
        onUpdate={onUpdate}
        ref={ref}
      />
    )
    const queryByNameAttribute = (name: string): HTMLElement | null => queryByAttribute('name', container, name)
    fireEvent.change(queryByNameAttribute('name')!, { target: { value: 'CV Step' } })

    fireEvent.click(getByText('cv.connectors.cdng.defineVerificationJob'))
    await waitFor(() => {
      expect(getByText('cv.connectors.cdng.jobName')).toBeTruthy()
    })

    fireEvent.click(getByText('cv.connectors.cdng.configureVerificationJob'))
    await waitFor(() => {
      expect(getByText('cv.connectors.cdng.selectTheJobNameFirst')).toBeTruthy()
    })
  })
})
