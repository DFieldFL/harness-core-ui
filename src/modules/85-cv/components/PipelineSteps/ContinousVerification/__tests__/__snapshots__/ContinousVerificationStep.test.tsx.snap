// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ContinousVerificationStep Step renders empty input sets 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <form
      class="FormikForm--main"
    >
      <div />
    </form>
    <p
      class="StyledProps--font StyledProps--main"
    >
      Errors
    </p>
    <pre>
      {}
    </pre>
    <button
      class="bp3-button StyledProps--font Button--button Button--with-current-color StyledProps--main StyledProps--intent-primary"
      type="button"
    >
      <span
        class="bp3-button-text"
      >
        Submit
      </span>
    </button>
  </div>
</div>
`;

exports[`Test ContinousVerificationStep Step renders input sets 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <form
      class="FormikForm--main"
    >
      <div>
        <div
          class="bp3-form-group"
          data-id="spec.sensitivity-0"
        >
          <label
            class="bp3-label"
          >
            sensitivity
             
            <span
              class="bp3-text-muted"
            />
          </label>
          <div
            class="bp3-form-content"
          >
            <div
              class="bp3-form-group"
              data-id="spec.sensitivity-1"
            >
              <div
                class="bp3-form-content"
              >
                <div
                  class="bp3-input-group"
                >
                  <input
                    autocomplete="off"
                    class="bp3-input"
                    name="spec.sensitivity"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="bp3-form-group"
          data-id="spec.duration-2"
        >
          <label
            class="bp3-label"
          >
            cv.connectors.cdng.duration
             
            <span
              class="bp3-text-muted"
            />
          </label>
          <div
            class="bp3-form-content"
          >
            <div
              class="bp3-form-group"
              data-id="spec.duration-3"
            >
              <div
                class="bp3-form-content"
              >
                <div
                  class="bp3-input-group"
                >
                  <input
                    autocomplete="off"
                    class="bp3-input"
                    name="spec.duration"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="bp3-form-group"
          data-id="spec.baseline-4"
        >
          <label
            class="bp3-label"
          >
            cv.connectors.cdng.baseline
             
            <span
              class="bp3-text-muted"
            />
          </label>
          <div
            class="bp3-form-content"
          >
            <div
              class="bp3-form-group"
              data-id="spec.baseline-5"
            >
              <div
                class="bp3-form-content"
              >
                <div
                  class="bp3-input-group"
                >
                  <input
                    autocomplete="off"
                    class="bp3-input"
                    name="spec.baseline"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
    <p
      class="StyledProps--font StyledProps--main"
    >
      Errors
    </p>
    <pre>
      {}
    </pre>
    <button
      class="bp3-button StyledProps--font Button--button Button--with-current-color StyledProps--main StyledProps--intent-primary"
      type="button"
    >
      <span
        class="bp3-button-text"
      >
        Submit
      </span>
    </button>
  </div>
</div>
`;

exports[`Test ContinousVerificationStep Step should render edit view as edit step 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <div
      class="Accordion--accordion accordion"
    >
      <div
        class="Accordion--panel"
        data-open="true"
        data-testid="step-1-panel"
      >
        <div
          class="Accordion--summary"
          data-testid="step-1-summary"
        >
          <div
            class="Accordion--label"
          >
            basic
          </div>
          <div
            class="Accordion--chevron"
          />
        </div>
        <div
          class="bp3-collapse Accordion--collapse"
          style="height: auto; overflow-y: visible; transition: none;"
        >
          <div
            aria-hidden="false"
            class="bp3-collapse-body"
            style="transform: translateY(0); transition: none;"
          >
            <div
              class="Accordion--details"
              data-testid="step-1-details"
            >
              <div
                class="formGroup md"
              >
                <div
                  class="InputWithIdentifier--txtNameContainer"
                >
                  <div
                    class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main InputWithIdentifier--txtIdContainer"
                  >
                    <p
                      class="StyledProps--font StyledProps--main"
                    >
                      ID
                      :
                    </p>
                    <span
                      class="bp3-popover-wrapper"
                    >
                      <span
                        class="bp3-popover-target"
                      >
                        <div
                          class="bp3-editable-text InputWithIdentifier--idValue"
                          tabindex="0"
                        >
                          <span
                            class="bp3-editable-text-content"
                            style="height: 1px; line-height: 1px; min-width: 0;"
                          >
                            ContinousVerification
                          </span>
                        </div>
                      </span>
                    </span>
                    <span
                      class="bp3-icon bp3-icon-edit StyledProps--main"
                      icon="edit"
                      style="vertical-align: middle;"
                    >
                      <svg
                        data-icon="edit"
                        height="12"
                        viewBox="0 0 16 16"
                        width="12"
                      >
                        <desc>
                          edit
                        </desc>
                        <path
                          d="M3.25 10.26l2.47 2.47 6.69-6.69-2.46-2.48-6.7 6.7zM.99 14.99l3.86-1.39-2.46-2.44-1.4 3.83zm12.25-14c-.48 0-.92.2-1.24.51l-1.44 1.44 2.47 2.47 1.44-1.44c.32-.32.51-.75.51-1.24.01-.95-.77-1.74-1.74-1.74z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </div>
                  <div
                    class="bp3-form-group"
                  >
                    <label
                      class="bp3-label"
                      for="name"
                    >
                      pipelineSteps.stepNameLabel
                       
                      <span
                        class="bp3-text-muted"
                      />
                    </label>
                    <div
                      class="bp3-form-content"
                    >
                      <div
                        class="bp3-input-group"
                      >
                        <input
                          autocomplete="off"
                          class="bp3-input"
                          name="name"
                          type="text"
                          value="CV"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="Accordion--panel"
        data-open="false"
        data-testid="step-2-panel"
      >
        <div
          class="Accordion--summary"
          data-testid="step-2-summary"
        >
          <div
            class="Accordion--label"
          >
            cv.connectors.cdng.defineVerificationJob
          </div>
          <div
            class="Accordion--chevron"
          />
        </div>
        <div
          class="bp3-collapse Accordion--collapse"
        >
          <div
            aria-hidden="false"
            class="bp3-collapse-body"
            style="transform: translateY(-undefinedpx);"
          />
        </div>
      </div>
      <div
        class="Accordion--panel"
        data-open="false"
        data-testid="step-3-panel"
      >
        <div
          class="Accordion--summary"
          data-testid="step-3-summary"
        >
          <div
            class="Accordion--label"
          >
            cv.connectors.cdng.configureVerificationJob
          </div>
          <div
            class="Accordion--chevron"
          />
        </div>
        <div
          class="bp3-collapse Accordion--collapse"
        >
          <div
            aria-hidden="false"
            class="bp3-collapse-body"
            style="transform: translateY(-undefinedpx);"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test ContinousVerificationStep Step should render edit view as new step 1`] = `
<div>
  <div
    class="OverlaySpinner--overlaySpinner"
  >
    <div
      class="Accordion--accordion accordion"
    >
      <div
        class="Accordion--panel"
        data-open="true"
        data-testid="step-1-panel"
      >
        <div
          class="Accordion--summary"
          data-testid="step-1-summary"
        >
          <div
            class="Accordion--label"
          >
            basic
          </div>
          <div
            class="Accordion--chevron"
          />
        </div>
        <div
          class="bp3-collapse Accordion--collapse"
          style="height: auto; overflow-y: visible; transition: none;"
        >
          <div
            aria-hidden="false"
            class="bp3-collapse-body"
            style="transform: translateY(0); transition: none;"
          >
            <div
              class="Accordion--details"
              data-testid="step-1-details"
            >
              <div
                class="formGroup md"
              >
                <div
                  class="InputWithIdentifier--txtNameContainer"
                >
                  <div
                    class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main InputWithIdentifier--txtIdContainer"
                  >
                    <p
                      class="StyledProps--font StyledProps--main"
                    >
                      ID
                      :
                    </p>
                    <span
                      class="bp3-popover-wrapper"
                    >
                      <span
                        class="bp3-popover-target"
                      >
                        <div
                          class="bp3-editable-text bp3-editable-text-placeholder InputWithIdentifier--idValue"
                          tabindex="0"
                        >
                          <span
                            class="bp3-editable-text-content"
                            style="height: 1px; line-height: 1px; min-width: 0;"
                          />
                        </div>
                      </span>
                    </span>
                    <span
                      class="bp3-icon bp3-icon-edit StyledProps--main"
                      icon="edit"
                      style="vertical-align: middle;"
                    >
                      <svg
                        data-icon="edit"
                        height="12"
                        viewBox="0 0 16 16"
                        width="12"
                      >
                        <desc>
                          edit
                        </desc>
                        <path
                          d="M3.25 10.26l2.47 2.47 6.69-6.69-2.46-2.48-6.7 6.7zM.99 14.99l3.86-1.39-2.46-2.44-1.4 3.83zm12.25-14c-.48 0-.92.2-1.24.51l-1.44 1.44 2.47 2.47 1.44-1.44c.32-.32.51-.75.51-1.24.01-.95-.77-1.74-1.74-1.74z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </div>
                  <div
                    class="bp3-form-group"
                  >
                    <label
                      class="bp3-label"
                      for="name"
                    >
                      pipelineSteps.stepNameLabel
                       
                      <span
                        class="bp3-text-muted"
                      />
                    </label>
                    <div
                      class="bp3-form-content"
                    >
                      <div
                        class="bp3-input-group"
                      >
                        <input
                          autocomplete="off"
                          class="bp3-input"
                          name="name"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="Accordion--panel"
        data-open="false"
        data-testid="step-2-panel"
      >
        <div
          class="Accordion--summary"
          data-testid="step-2-summary"
        >
          <div
            class="Accordion--label"
          >
            cv.connectors.cdng.defineVerificationJob
          </div>
          <div
            class="Accordion--chevron"
          />
        </div>
        <div
          class="bp3-collapse Accordion--collapse"
        >
          <div
            aria-hidden="false"
            class="bp3-collapse-body"
            style="transform: translateY(-undefinedpx);"
          />
        </div>
      </div>
      <div
        class="Accordion--panel"
        data-open="false"
        data-testid="step-3-panel"
      >
        <div
          class="Accordion--summary"
          data-testid="step-3-summary"
        >
          <div
            class="Accordion--label"
          >
            cv.connectors.cdng.configureVerificationJob
          </div>
          <div
            class="Accordion--chevron"
          />
        </div>
        <div
          class="bp3-collapse Accordion--collapse"
        >
          <div
            aria-hidden="false"
            class="bp3-collapse-body"
            style="transform: translateY(-undefinedpx);"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
